service/src/main/kotlin/xyz/lebkuchenfm/external/storage/mongo/repositories/LlmSituationPromptsMongoRepository.kt

```diff
@@ -0,0 +1,47 @@
+package xyz.lebkuchenfm.external.storage.mongo.repositories
+
+import com.mongodb.client.model.Filters
+import com.mongodb.client.model.Sorts
+import com.mongodb.kotlin.client.coroutine.MongoDatabase
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.datetime.Instant
+import kotlinx.serialization.Contextual
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import org.bson.types.ObjectId
+import xyz.lebkuchenfm.domain.radiopersonality.llmprompts.LlmPromptCreation
+import xyz.lebkuchenfm.domain.radiopersonality.llmprompts.LlmSituationPrompt
+import xyz.lebkuchenfm.domain.radiopersonality.llmprompts.LlmSituationPromptsRepository
+import xyz.lebkuchenfm.domain.radiopersonality.llmprompts.LlmSituationType
+import xyz.lebkuchenfm.domain.users.UserName
+
+class LlmSituationPromptsMongoRepository(database: MongoDatabase) : LlmSituationPromptsRepository {
+    private val collection = database.getCollection<LlmSituationPromptEntity>("llmSituationPrompts")
+
+    override suspend fun findLatestByType(type: LlmSituationType): LlmSituationPrompt? {
+        return collection
+            .find(Filters.eq(LlmSituationPromptEntity::type.name, type.name))
+            .sort(Sorts.descending(LlmSituationPromptEntity::createdAt.name))
+            .limit(1)
+            .firstOrNull()
+            ?.toDomain()
+    }
+}
+
+@Serializable
+private data class LlmSituationPromptEntity(
+    @Contextual @SerialName("_id") val id: ObjectId? = null,
```

Comment:
🤔  do we even need that id here?

Reply:
We don't. I was wondering about this one too. In many Mongo entities we have `id` field, even though we don't ever use it. I've added this because that's what that document actually looks like in Mongo. If we want to remove the `id` here, then we should also remove it everywhere else.

Reply:
+1 for removing everywhere